src/Pelagos/Bundle/AppBundle/Command/ReportDoiStatusDatasetStatusCommand.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+namespace Pelagos\Bundle\AppBundle\Command;
+
+use Exception;
+
+use Pelagos\Entity\Dataset;
+use Pelagos\Entity\DOI;
+use Pelagos\Util\DOIutil;
+
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Output\StreamOutput;
+
+use Doctrine\ORM\EntityManager;
+
+/**
+ * This Symfony Command generates a report for DOI migration.
+ *
+ * @see ContainerAwareCommand
+ */
+class ReportDoiStatusDatasetStatusCommand extends ContainerAwareCommand
+{
+    /**
+     * A name of the file that will store the results of this report program.
+     *
+     * It is initialized to a default value.
+     *
+     * @var string
+     */
+    protected $outputFileName = '';
+
+    /**
+     * The Symfony Console output object.
+     *
+     * @var OutputInterface fileOutput
+     */
+    protected $fileOutput = null;
+
+    /**
+     * The Doctrine entity manager - ORM critter.
+     *
+     * @var EntityManager entityManager
+     */
+    protected $entityManager;
+
+    /**
+     * The file output array which stores the data.
+     *
+     * @var array
+     */
+    protected $fileOutputArray;
+
+    /**
+     * Configuration for the command script.
+     *
+     * @return void
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('reports:doi-status-datasets')
+            ->setDescription('Report of udi(s) with doi status, dataset status')
+            ->addArgument('outputFileName', InputArgument::REQUIRED, 'What is the output file path and name?');
+    }
+
+    /**
+     * Executes the current command.
+     *
+     * @param InputInterface  $input  An InputInterface instance.
+     * @param OutputInterface $output An OutputInterface instance.
+     *
+     * @throws Exception Exception thrown when openIO function fails to generate report.
+     *
+     * @return integer Return 0 on success, or an error code otherwise.
```

Comment:
The return is always only 0. There are no other returns here. The fact that it throws an error is described in @throws.